world.lua

```diff
@@ -243,7 +249,9 @@ function World:newCollider(collider_type, shape_arguments, table_to_use)
 
    -- index by self for now
    o._world = self
+   table.insert(self.colliders, o)
    self.colliders[o] = o
```

Comment:
This seems to add the same table twice to the list of colliders. This is most likely a double-free bug.

Reply:
I think the reason the bug occurs is because in `Collider:destroy` it's destroying the collider through indexing by itself, but the collider is still up and dandy as an array index.

I made a fix and it *seems* to work fine:
```lua
function Collider:destroy()
   self._world.colliders[self] = nil
   for i, v in ipairs(self._world.colliders) do
      if v == self then
         table.remove(self._world.colliders, i)
         break
      end
   end
   self._world.colliders[self] = nil
   self.fixture:setUserData(nil)
   self.fixture:destroy()
   self.body:destroy()
end
```

If you're ok with having gaps in `self._world.colliders`, you could do `self._world.colliders[i] = nil` instead of `table.remove`, but it wouldn't re-order the table indices.